tests/services/storage/test_archive.py

```diff
@@ -199,236 +203,48 @@ def test_create_restore_session_with_encryption_key_calls_post_with_encryption_k
     def test_start_restore_calls_post_with_correct_url(self, connection):
         storage_archive_service = StorageArchiveService(connection)
 
-        storage_archive_service.start_restore(
-            DEVICE_GUID, WEB_RESTORE_SESSION_ID, PATH_SET, NUM_FILES, NUM_DIRS, SIZE
-        )
-        assert connection.post.call_args[ARGS_INDEX][ARGS_INDEX] == WEB_RESTORE_JOB_URL
-
-    def test_start_restore_with_required_args_calls_post_with_all_args_in_json(
-        self, connection
-    ):
-        storage_archive_service = StorageArchiveService(connection)
-        storage_archive_service.start_restore(
-            DEVICE_GUID, WEB_RESTORE_SESSION_ID, PATH_SET, NUM_FILES, NUM_DIRS, SIZE
-        )
-
-        json_arg = connection.post.call_args[KWARGS_INDEX][JSON_KEYWORD]
-
-        keys = [
-            GUID_KEY,
-            WEB_RESTORE_SESSION_ID_KEY,
-            PATH_SET_KEY,
-            NUM_FILES_KEY,
-            NUM_DIRS_KEY,
-            SIZE_KEY,
-            ZIP_RESULT_KEY,
-            EXPIRE_JOB_KEY,
-            SHOW_DELETED_KEY,
-            RESTORE_FULL_PATH_KEY,
-            TIMESTAMP_KEY,
-            BACKUP_SET_ID_KEY,
-        ]
-
-        assert sorted(json_arg.keys()) == sorted(keys)
-
-    def test_start_restore_with_opt_zip_result_as_false_calls_post_with_zip_result_in_data(
-        self, connection
-    ):
-        storage_archive_service = StorageArchiveService(connection)
-
         storage_archive_service.start_restore(
             DEVICE_GUID,
             WEB_RESTORE_SESSION_ID,
-            PATH_SET,
+            RESTORE_GROUPS,
             NUM_FILES,
             NUM_DIRS,
-            SIZE,
-            zip_result=False,
+            NUM_BYTES,
         )
-        json_arg = connection.post.call_args[KWARGS_INDEX][JSON_KEYWORD]
-        assert json_arg.get(ZIP_RESULT_KEY) is False
-
-    def test_start_restore_with_opt_zip_result_as_true_calls_post_with_zip_result_in_data(
-        self, connection
-    ):
-        storage_archive_service = StorageArchiveService(connection)
-
-        storage_archive_service.start_restore(
-            DEVICE_GUID,
-            WEB_RESTORE_SESSION_ID,
-            PATH_SET,
-            NUM_FILES,
-            NUM_DIRS,
-            SIZE,
-            zip_result=True,
+        assert (
```

Comment:
I deleted a lot of tests here because they were testing each parameter individually and we don't typically do that in our tests, we usually just do `tests_calls_url_with_expected_data` and it just tests all the params and the url in the same test... which is much easier to maintain.

@kiran-chaudhary here is where the URL gets tested for the updated call

Reply:
View raw file